ucm2/xiaomi-mido/HiFi.conf

```diff
@@ -0,0 +1,46 @@
+# Use case configuration for Xiaomi Redmi Note 4
+# All analog outputs/inputs connected to internal PM8953 codec,
+# speaker connected via AW87318CSR amplifier to Lineout.
+
+SectionVerb {
+	Value {
+		PlaybackPCM "plughw:${CardId},0"
+		CapturePCM "plughw:${CardId},1"
+	}
+}
+<platforms/msm8916/qdsp6-components.conf>
+
+SectionDevice."Speaker" {
+        Comment "Speaker playback"
+
+        ConflictingDevice [
+                "Headphones"
+                "Earpiece"
+        ]
+
+        EnableSequence [
+                cset "name='LINEOUT' 1"
+                cset "name='LINEOUT Ext' 1"
```

Comment:
I think you should be able to remove this if you change in `audio-routing`:

```
"Speaker Amp IN", "LINEOUT_OUT",
```

then setting `LINEOUT` to `1` like you do one line above should be sufficient. I don't see any reason for adding the special Ext stuff here.